source/common/common/mutex_tracer.cc

```diff
@@ -0,0 +1,39 @@
+#include "common/common/mutex_tracer.h"
+
+#include <iostream>
+#include <memory>
+
+#include "common/common/assert.h"
+
+#include "absl/synchronization/mutex.h"
+
+namespace Envoy {
+
+MutexTracer* MutexTracer::singleton_ = nullptr;
+
+MutexTracer* MutexTracer::getOrCreateTracer() {
+  if (singleton_ == nullptr) {
+    singleton_ = new MutexTracer;
+    absl::RegisterMutexTracer(&Envoy::MutexTracer::contentionHook);
```

Comment:
Do we need to clean this up at exit or during tests? Not sure how this works under the hood.

Reply:
Yeah this is once per process, so I guess that means all tests run after the MutexTracer run will be content-tracked whether they want it or not.

The question is: is there a way in absl to unregister the handler...I don't see one doc'd in https://github.com/abseil/abseil-cpp/blob/master/absl/synchronization/mutex.h and from looking at the code I don't know if passing in null for the function-pointer would work. But maybe we could unregister by passing in a function that did nothing.  OTOH what our default function does is pretty innocent; it just bumps some counters, so maybe we can live with this for tests?

Reply:
Yeah that's probably fine. I would just add some comments.